### PR TITLE
Index limits

### PR DESCRIPTION
**Pull request by benthorner**
https://trello.com/c/3OXJzoo5/206-limits-and-filtering-on-edit-content-page

This ensures we cleanup (no newlines) and truncate the title and summary
before we save them, if the user ignores the guidance for these fields.

For the base_path, we similarly cleanup and truncate the title before
passing it to the path generator service, thus limiting the slug length.

### CODE REVIEW

File: app/controllers/documents_controller.rb [hunk 1]
```diff
@@ -58,10 +58,7 @@ def filter_params
   end
 
   def update_params(document)
-    contents_params = document.document_type_schema.contents.map(&:id)
-    base_path = PathGeneratorService.new.path(document, params[:document][:title])
-
-    params.require(:document).permit(:title, :summary, :update_type, :change_note, contents: contents_params)
-      .merge(base_path: base_path, publication_state: "changes_not_sent_to_draft", review_state: "unreviewed")
+    DocumentAttributes.new(document).attributes(params)
```
Comment (kevindew) [inline app/controllers/documents_controller.rb]: Do you think this should be called something with a reference to update params?

File: app/services/document_attributes.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+  attr_reader :document
+
+  def initialize(document)
+    @document = document
+  end
+
+  def attributes(params)
+    path_title = params[:document][:title].chomp[0...TITLE_SLUG_MAX_LENGTH]
+    base_path = PathGeneratorService.new.path(document, path_title)
```
Comment (kevindew) [inline app/services/document_attributes.rb]: What happens if title is nil?

File: app/services/document_attributes.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  SUMMARY_MAX_LENGTH = 600
+
+  attr_reader :document
+
+  def initialize(document)
+    @document = document
+  end
+
+  def attributes(params)
+    path_title = params[:document][:title].chomp[0...TITLE_SLUG_MAX_LENGTH]
```
Comment (kevindew) [inline app/services/document_attributes.rb]: We could probably do a `params.require(:document)` so we don't need to nest into params each time

File: app/services/document_attributes.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    base_path = PathGeneratorService.new.path(document, path_title)
+
+    title = params[:document][:title].chomp[0...TITLE_MAX_LENGTH]
```
Comment (kevindew) [inline app/services/document_attributes.rb]: I think chomp only strips new lines at the end of strings?

File: app/controllers/documents_controller.rb (hunk 1 above)
Comment (benthorner) [reply app/controllers/documents_controller.rb]: Renamed to DocumentUpdateParams.

File: app/services/document_attributes.rb (hunk 3 above)
Comment (benthorner) [reply app/services/document_attributes.rb]: OK - you win ;-).

File: app/services/document_attributes.rb (hunk 4 above)
Comment (benthorner) [reply app/services/document_attributes.rb]: Fixed - my misunderstanding.

File: app/services/document_attributes.rb (hunk 2 above)
Comment (benthorner) [reply app/services/document_attributes.rb]: Coping now - good spot :-).

File: app/services/document_update_params.rb [hunk 5]
```diff
@@ -12,11 +12,11 @@ def initialize(document)
   end
 
   def update_params(params)
-    path_title = params[:document][:title].chomp[0...TITLE_SLUG_MAX_LENGTH]
+    path_title = params[:document][:title].squish[0...TITLE_SLUG_MAX_LENGTH]
```
Comment (emmabeynon) [inline app/services/document_update_params.rb]: I've not seen `squish` before but I like it 😄

Comment (kevindew) [reply app/services/document_update_params.rb]: Oh yes that's a new one for me too

**Pull request merged**